salt/states/boto3_route53.py

```diff
@@ -798,3 +826,204 @@ def rr_absent(name, HostedZoneId=None, DomainName=None, PrivateZone=False,
+                update = True
+                break
+            if ((ChildHealthChecks is not None) and (check.get('ChildHealthChecks') is not None)):
+                if sorted(ChildHealthChecks) != sorted(check.get('ChildHealthChecks')):
+                    update = True
+            if ((Regions is not None) and (check.get('Regions') is not None)):
+                if sorted(Regions) != sorted(check.get('Regions')):
+                    update = True
+            if ((AlarmIdentifier is not None) and (check.get('AlarmIdentifier') is not None)):
+                if AlarmIdentifier.get('Name') != check.get('AlarmIdentifier', {}).get('Name'):
+                    update = True
+                if AlarmIdentifier.get('Region') != check.get('AlarmIdentifier', {}).get('Region'):
+                    update = True
+        for key in resettables:
+            if params.get(key) is None and check.get(key) is not None:
+                update = True
+                reset_elements.append(key)
+
+    if not create and not update:
+        ret['comment'] = 'Route 53 health check {} is already in the desired state.'.format(Name)
+        log.info(ret['comment'])
+        return ret
+    elif create:
+        keys = [
+            'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+            'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+            'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks', 'CallerReference'
+        ]
+
+        items = {key: params.get(key) for key in keys if params.get(key) is not None}
+        args = {'Name': Name, 'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
+        args.update(items)
+        log.debug(' ==> {}'.format(items))
+
+        if __opts__['test']:
+            ret['comment'] = 'Route 53 health check {} would be created.'.format(Name)
+            ret['changes'] = items
+            ret['result'] = None
+            return ret
+
+        check = __salt__['boto3_route53.create_health_check'](**args)
+        if check:
+            ret['comment'] = 'Route 53 health check {} successfully created.'.format(Name)
+            log.info(ret['comment'])
+            ret['changes']['new'] = check
+        else:
+            ret['comment'] = 'Creation of Route 53 health check {} failed.'.format(Name)
+            log.error(ret['comment'])
+            ret['result'] = False
+            return ret
+    elif update:
+        keys = [
+            'HealthCheckVersion', 'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+            'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+            'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks', 'CallerReference'
+        ]
```

Comment:
It looks like you've got this same keys collection multiple times - is there some better way to do this, maybe by creating a dictionary or using `**kwargs` or something?

Reply:
I tried, but there are different sub-sets of keys which should be handled on each case, and after quite many errors (removing/adding the wrong key here and there), I ended up separating each key-set, as to make it easier to understand what is used/required on each case.